src/ingest-pipeline/airflow/dags/diagnostics/plugins/illegal_argument_plugin.py

```diff
@@ -0,0 +1,55 @@
+"""
+This plugin uses regex to search for IllegalArgument errors in session.log
+"""
+
+import re
+from itertools import dropwhile, groupby
+
+from diagnostics.diagnostic_plugin import DiagnosticResult, RegexDiagnosticPlugin
+
+"""
+DOCKER_REGEX locates the section of session.log where both docker containers are started
+and captures everything after that, inclusive of the container start log, as a rough
+way of logging an error taking place in this type of docker container
+"""
+DOCKER_REGEX = r"job collect_ometiff_files.*docker.*job convert_to_pyramid.*docker*.*"
```

Comment:
Hey, I've got an idea.  Can this be generalized to catch *any* docker run?  They all start with 'docker ... run' and end with something like "[job ...] Max memory used".  Your new class for detecting docker errors would be a lot more useful if it was more general.
```

Reply:
Gave this a shot, please see RegexDiagnosticPlugin class.